aria-practices.html

```diff
@@ -3697,6 +3697,1247 @@ <h5>Examples</h5>
+          <li>
+            <p>Finally, if the name is still empty, then other fallback host-language-specific attributes or elements are used if present. For HTML, these are, depending on the element:</p>
+            <dl class="switch">
+              <dt><code>input</code> whose <code>type</code> attribute is in the Text, Password, Search, Telephone, or URL states</dt>
+              <dt><code>textarea</code></dt>
+              <dd>The <code>title</code> attribute. Otherwise, the <code>placeholder</code> attribute.</dd>
+
+              <dt><code>input</code> whose <code>type</code> attribute is in the Submit Button state</dt>
+              <dd>A localized string of the word "submit".</dd>
+
+              <dt><code>input</code> whose <code>type</code> attribute is in the Reset Button state</dt>
+              <dd>A localized string of the word "reset".</dd>
+
+              <dt><code>input</code> whose <code>type</code> attribute is in the Image Button state</dt>
+              <dd>The <code>title</code> attribute. Otherwise, a localized string of the phrase "Submit Query".</dd>
+
+              <dt><code>summary</code></dt>
+              <dd>The word "Details".</dd>
+
+              <dt>Other elements</dt>
+              <dd>The <code>title</code> attribute.</dd>
+            </dl>
+          </li>
+        </ol>
+        <p>
+          The final step is a fallback mechanism. Generally when labelling an element, use one of the non-fallback mechanisms.
+        </p>
+        <p>
+          When calculating a name from content, the user agent walks through all descendant nodes.
+          And, when following <code>aria-labelledby</code> references, it similarly walks the tree of each referenced element.
+          Thus, the naming algorithm is recursive.
+          The following two sections provide non-recursive and recursive examples of how the algorithm works.
+        </p>
+
+        <section id="name_calculation_non-recursive_ex">
+          <h5>Examples of non-recursive accessible name calculation</h5>
+          <p>Consider an <code>input</code> element that has no associated <code>label</code> element and only a <code>name</code> attribute and so does not have an accessible name (do not do this):</p>
+          <pre><code>&lt;input name="code"></code></pre>
+          <p>If there is a <code>placeholder</code> attribute, then it serves as a naming falback mechanism (avoid doing this):</p>
+
+          <pre><code>&lt;input name="code"
+       placeholder="One-time code"></code></pre>
+
+          <p>If there is also a <code>title</code> attribute, then it is used as the accessible name instead of <code>placeholder</code>, but it is still a fallback (avoid doing this):</p>
+
+          <pre><code>&lt;input name="code"
+       placeholder="123456"
+       title="One-time code"></code></pre>
+
+          <p>If the
```

Comment:
Instead of:
> then that overrides the label element

How about the following, for consistency with previous examples:
> then that becomes the accessible name, overriding the label element

Also, consider adding a disclaimer in brackets, i.e., something like:
> If there is also an aria-label attribute (not recommended unless it adds clarity for assistive technology users), then ...